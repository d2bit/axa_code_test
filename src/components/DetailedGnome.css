.modal {
  z-index: 99;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(128, 128, 128, 0.8);
  display: flex;
}

.frame {
  background-color: whitesmoke;
  margin: 3rem auto;
  padding: 1rem 2rem;
  flex-basis: 100%;
  align-self: center;
  border: 1px solid grey;
  border-radius: 50px 19px;
}

.thumbnail {
  width: 100%;
  max-height: 200px;
  height: auto;
  border: 1px solid black;
}

.listTitle {
  text-transform: capitalize;
  margin: 1rem 0 0.5rem;
}
.list {
  list-style-type: none;
  margin: 0;
  padding: 0;
}
.listItem {
  padding: .3rem 1rem;
  display: inline-block;
  border: 1px solid;
  border-radius: 10px;
  margin: .3rem;
}

.details {
}
.detailsTitle {
  display: inline-block;
  width: 40%;
  margin: 0;
  font-weight: 600;
  text-transform: capitalize;
}
.detailsValue {
  display: inline-block;
  width: 60%;
  margin: 0;
}

@media (min-width: 550px) {
  .frame {
    flex-basis: 60%;
  }
}

@media (min-width: 850px) {
  .frame {
    flex-basis: 40%;
  }
}
