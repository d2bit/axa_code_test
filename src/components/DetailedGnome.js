import React from 'react'

const DetailedGnome = ({ id, name, thumbnail, age, weitght, height, hair_color, professions, friends, closeFn }) => {
  return (
    <div data-testid={`DetailedGnome-${id}`}>
      <h3 data-testid='name'>{name}</h3>
      <img width='200px' data-testid='thumbnail' src={thumbnail} alt={name} />
      <button onClick={closeFn}>CLOSE</button>
    </div>
  )
}

export default DetailedGnome
