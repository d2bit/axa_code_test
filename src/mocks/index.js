import data from './data.json'

const population = Object.values(data)[0]
const populationCount = population.length

export function getPopulation(size=5) {
  if (size >= populationCount) {
    return population
  }
  return population.slice(0, size)
}

export function getGnome(index) {
  if (index && index >= 0 && index < populationCount) {
    return population[index]
  }
  const randomIndex = Math.floor(Math.random() * populationCount)
  return population[randomIndex]
}

export function getResponseData() {
  return data
}

const Helper = {
  getPopulation,
  getGnome,
  getResponseData,
}

export default Helper
