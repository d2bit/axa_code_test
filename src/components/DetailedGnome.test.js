import React from 'react'
import { render } from 'react-testing-library'
import 'dom-testing-library/extend-expect'
import { getGnome } from '../mocks'
import DetailedGnome from './DetailedGnome'

describe('DetailedGnome', () => {
  it('shows the gnome information', () => {
    const gnomeId = 13
    const gnomeProfile = getGnome(gnomeId)

    const { getByTestId, container } = render(<DetailedGnome {...gnomeProfile} />)

    expect(getByTestId('name')).toHaveTextContent(gnomeProfile.name)
    expect(getByTestId('thumbnail')).toHaveAttribute('src', gnomeProfile.thumbnail)
    expect(container).toMatchSnapshot()
  })
})
