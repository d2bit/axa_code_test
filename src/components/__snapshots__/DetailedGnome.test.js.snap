// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedGnome shows the gnome information 1`] = `
<div>
  <div
    data-testid="DetailedGnome-13"
  >
    <article>
      <h1
        data-testid="name"
      >
        Tobus Nozzletink
      </h1>
      <img
        alt="Tobus Nozzletink"
        data-testid="thumbnail"
        src="http://www.publicdomainpictures.net/pictures/120000/velka/cactus-1430656432Iug.jpg"
      />
      <dl
        data-testid="Details"
      >
        <dt>
          age
        </dt>
        <dd>
          367
        </dd>
        <dt>
          weight
        </dt>
        <dd>
          36.732063
        </dd>
        <dt>
          height
        </dt>
        <dd>
          129.37299
        </dd>
        <dt>
          hair color
        </dt>
        <dd>
          Green
        </dd>
      </dl>
      <h3>
        professions
      </h3>
      <ul
        data-testid="professions"
      >
        <li>
           Tinker
        </li>
        <li>
          Metalworker
        </li>
        <li>
          Carpenter
        </li>
      </ul>
      <h3>
        friends
      </h3>
      <ul
        data-testid="friends"
      >
        <li>
          Milli Clanksbuster
        </li>
        <li>
          Kinthony Tinkrivet
        </li>
        <li>
          Sarabink Voidbuster
        </li>
        <li>
          Tobus Nozzletorque
        </li>
      </ul>
    </article>
  </div>
</div>
`;
